qemu/tests/qemu_guest_agent.py

```diff
@@ -2686,6 +2686,63 @@ def bl_check(qga_cmd):
             cmd = "mv -f /etc/sysconfig/qemu-ga-bk /etc/sysconfig/qemu-ga"
             session.cmd(cmd)
 
+    @error_context.context_aware
+    def gagent_check_virtio_device(self, test, params, env):
+        """
+        Get hostname, timezone and currently active users on the vm.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        def _result_check(rsult_qga, rsult_guest):
+            if rsult_qga != rsult_guest:
+                msg = "The result is different between qga and guest\n"
+                msg += "from qga: %s\n" % rsult_qga
+                msg += "from guest: %s\n" % rsult_guest
+                test.fail(msg)
+
+        devs_list = self.gagent.get_virtio_device()
+        for device in devs_list:
+            driver_name = device["driver-name"]
+            error_context.context("Check %s info." % driver_name, logging.info)
+
+            driver_date = device["driver-date"]
+            driver_version = device["driver-version"]
+            device_address = device["address"]["data"]
+            device_id = device_address["device-id"]
+            vendor_id = device_address["vendor-id"]
+            check_driver_cmd = params["check_driver_powershell_cmd"]
```

Comment:
better put LINE 2718 out of this `for` loop.

Reply:
update